### ISSUE TITLE
Partial resists on Frostbolt and Cone of Cold (binary spells) [spell]

### ISSUE THREAD

**Issue by knasen88** [Atlantiss/NetherwingBugtracker#5013]
[//]: # (Enclose links to things related to the bug using http://wowhead.com or any other TBC database.)
[//]: # (You can use screenshot ingame to visual the issue.)
[//]: # (Write your tickets according to the format:)
[//]: # ([Quest][Azuremyst Isle] Red Snapper - Very Tasty!)
[//]: # ([NPC] Magistrix Erona)
[//]: # ([Spell][Mage] Fireball)
[//]: # ([Npc][Drop] Ghostclaw Lynx)
[//]: # ([Web] Armory doesnt work)
[//]: # (To find server revision type `.server info` command in the game chat.)

**Description**: 
getting partial resist on Frostbolt and Cone of Cold. Binary spell is hit or miss, no partial
**Current behaviour**:
partial on binary
**Expected behaviour**: 
all non dots with 2 effects (dmg & debuff) are generally binary
**Server Revision**:

**Comment by Wolffenstein** [comment]
Frostbolt is non-binary in TBC (or at least 2.3+).

**Comment by knasen88** [comment]
what about Cone of Cold then?

**Comment by knasen88** [comment]
in addition, a bit offtopic if you don't mind. The air elementals, Storming Wind-Ripper, what are their resistances? I got 73 spellpen available but I was still getting very high amount of resist on every school I had.

**Comment by Wolffenstein** [comment]
Fixed in rev 2795.

**Comment by ircfelix** [comment]
Still getting partial resists on frostbolts.

**Comment by ircfelix** [comment]
@Wolffenstein still getting partials on frostbolts.

**Comment by Easyfraggy** [comment]
He literally said above that Frostbolt is non-binary so you should get partial resists.

**Comment by ircfelix** [comment]
http://web.archive.org/web/20080913120526/http://elitistjerks.com/f31/t18441-mage_sweet_informational_thread/

> "Frostbolt, however, is a binary spell"

**Comment by ircfelix** [comment]
https://dwarfpriest.wordpress.com/2008/01/07/spell-hit-spell-penetration-and-resistances/

Posted well after patch 2.3:

> Binary and Non-Binary Spells

Spell Resistance functions slightly differently depending on what kind of spell you are casting. There are only two different kinds of spells in this regard: Binary and Non-Binary.

Binary spells as named as such as there are only two outcomes for them; they can either hit or miss. Spells that do no damage, or that have a snare effect built in to them (like Mind Flay or Frostbolt), are binary spells.

**Comment by ircfelix** [comment]
I imagine (but I could also be very wrong with this) that this fix would require only replacing spell type from non-binary to binary. Why do I bring this up? Because I believe this fix is easy and every Frost mage out there would be super thankful!

**Comment by ircfelix** [comment]
Could you please fix the frostbolt binary spell problem? https://github.com/Atlantiss/NetherwingBugtracker/issues/5013
The issue with it is that Frostbolts (don't know about cone of cold) hit for significantly less damage. Frostbolts in my gear and with consumables should hit for ~2793 on average (=normal hits and crits average, number taken from my arcane mage spreadsheet), however they hit for ~2590 on average (calculation see image: https://i.imgur.com/1dZtGny.jpg and 2015*0,657+4603*0,275=2590).
In other words, in the current state Frostbolts hit for **200** damage (on average) less than they should and I am not even specced deeper into frost and neither have a pure +spell damage gear, as a deep frost mage would have. So for deep frost mages the damage loss is even higher. This is kinda crucial for mages who use Frostbolt.

**Comment by Wolffenstein** [comment]
Fixed in rev 2844.

**Comment by ircfelix** [comment]
Observed several full resists in a row on bosses with coe, whilst having 14,16% hit (like 5 out of 8 being resisted). Is it possible, that Frostbolts which got partially resisted before the fix are now full resists?

**Comment by ircfelix** [comment]
Requesting Reopen (see post above)

**Comment by ircfelix** [comment]
Yesterday on curator I had tons of full frostbolt resists again. Much more than on any other boss. Either curator has a higher boss level or a higher resist to Frost.

**Comment by JeanCodeVanDamme** [comment]
> Fixed in rev 2844.



> Yesterday on curator I had tons of full frostbolt resists again. Much more than on any other boss. Either curator has a higher boss level or a higher resist to Frost.

There is still something wrong with frostbolts and its resists.
Even with hitcap, where you have a 1% chance it happens way too often in a single fight.
Please look this up again, because as it is right now I am pretty screwed with my choice to go with the frost dmg set as a tailor.

**Comment by Easyfraggy** [comment]
Would need some proof that there is a problem here. I did some casts on Curator today with 3% spell hit.

![image](https://user-images.githubusercontent.com/36134067/56595105-525d9f80-65ee-11e9-8e12-01e046ff1f8b.png)

It's only 1350 casts but I think it's only down to +-1% at this amount of casts. Seems pretty ok to me.

**Comment by ircfelix** [comment]
> 
> 
> Would need some proof that there is a problem here. I did some casts on Curator today with 3% spell hit.
> 
> ![image](https://user-images.githubusercontent.com/36134067/56595105-525d9f80-65ee-11e9-8e12-01e046ff1f8b.png)
> 
> It's only 1350 casts but I think it's only down to +-1% at this amount of casts. Seems pretty ok to me.

This was the The Curator try I noticed there are problems:
https://i.imgur.com/gR2kPP1.jpg
The SS was taken after the binary change of frostbolt was reverted (so frostbolt should not partial resist now) and which led me to believe something is wrong with frb. During that try I was at ~130 or 131 hit rating and had 3/3 elemental precision, which should give 6% hit according to Wolf.

**Comment by Easyfraggy** [comment]
I'll run another test without the talent to see if thats causing discrepancy.

**Comment by Easyfraggy** [comment]
Tested on Doomwalker with 16.09% hit.

![Uploading image.png…]()

**Comment by Easyfraggy** [comment]
Did anther 1200 on curator with 16.09% hit.

![image](https://user-images.githubusercontent.com/36134067/56691668-d001d800-66e0-11e9-8b51-ff3a221a1ce1.png)

**Comment by Easyfraggy** [comment]
If anything the talent only gives 3% hit not 6% - resist chance is working fine otherwise it seems.

**Comment by ircfelix** [comment]
The talent should give 6% for frost. @Wolffenstein fixed it.

**Comment by Easyfraggy** [comment]
Well thats not what this issue is about. You should report that in the proper thread if that is the case.

**Comment by ircfelix** [comment]
> 
> 
> Well thats not what this issue is about. You should report that in the proper thread if that is the case.

I don't even know if the talent is the problem and I didn't suspect it to be the problem.

**Comment by Easyfraggy** [comment]
I've just tested it for you so clearly frostbolt partial resists which this issue is about is not the problem. I even tested it on curator (although he has no extra resists). Without using spell penetration and without CoE. Which means this issue is invalid.
